bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/DMAssetPostProcessor.java

```diff
@@ -55,46 +55,53 @@
 
     @Override
     public void process(SlingHttpServletRequest request, List<Modification> list) throws Exception {
-        Modification lastFileReferenceModification = getLastPropertyModification(request, list, DownloadResource.PN_REFERENCE);
+
+        Modification lastFileReferenceModification = getLastPropertyModification(list, DownloadResource.PN_REFERENCE);
         if (lastFileReferenceModification != null) {
-            switch (lastFileReferenceModification.getType()) {
-                case CREATE:
-                case MODIFY:
-                    ValueMap valueMap = request.getResource().getValueMap();
-                    String fileReference = valueMap.get(DownloadResource.PN_REFERENCE, String.class);
-                    if (fileReference != null) {
-                        Resource assetResource = request.getResourceResolver().getResource(fileReference);
-                        if (assetResource != null) {
-                            Asset asset = assetResource.adaptTo(Asset.class);
-                            if (asset != null) {
-                                if(isDmAsset(asset)) {
-                                    String[] productionAssetUrls = publishUtils.externalizeImageDeliveryAsset(assetResource);
-                                    String imageServerUrl = productionAssetUrls[0] + IMAGE_SERVER_PATH;
-                                    checkSetProperty(request.getResource(), Image.PN_IMAGE_SERVER_URL, imageServerUrl, list);
-                                } else {
-                                    checkSetProperty(request.getResource(), Image.PN_IMAGE_SERVER_URL, null, list);
-                                }
+            String pathToComponent =  lastFileReferenceModification.getSource().split("/" + DownloadResource.PN_REFERENCE)[0];
```

Comment:
@oborysyu use substring instead of split to avoid regexp overhead

Reply:
done.